File: 20180710/Shortest Word Distance.py
class Solution(object):
    def shortestDistance(self, words, word1, word2):
        """
        :type words: List[str]
        :type word1: str
        :type word2: str
        :rtype: int
        """
        idx = -1
        res = len(words)
        for i, w in enumerate(words):
            if w == word1 or w == word2:
                if idx != -1 and words[idx] != w:
                    res = min(res, i - idx)
                idx = i
        return res

File: 20180710/Shortest Word Distance.java
class Solution {
    public int shortestDistance(String[] words, String word1, String word2) {
        int idx = -1;
        int res = words.length;
        for(int i = 0; i < words.length; i ++){
            if (words[i].equals(word1) || words[i].equals(word2)){
                if (idx != -1 && !words[idx].equals(words[i])){
                    res = Math.min(res, i - idx);
                }
                idx = i;
            }
        }
        return res;
    }
}

